shared/reports/resources.py

```diff
@@ -285,105 +252,31 @@ def append(self, _file, joined=True):
 
         assert _file.name, "file must have a name"
 
-        # check if file already exists
-        index = self._files.get(_file.name)
-        if index:
-            # existing file
-            # =============
-            #  merge old report chunk
-            cur_file = self[_file.name]
-            # merge it
-            cur_file.merge(_file, joined)
-            # set totals
-            index.file_totals = cur_file.totals
-            # update chunk in report
-            self._chunks[index.file_index] = cur_file
+        existing_file = self._files.get(_file.name)
+        if existing_file is not None:
+            existing_file.merge(_file, joined)
         else:
-            # new file
-            # ========
-            # override totals
-            if not joined:
-                _file._totals = ReportTotals(0, _file.totals.lines)
-
-            # add to network
-            self._files[_file.name] = ReportFileSummary(
-                len(self._chunks),  # chunk location
-                _file.totals,  # Totals
-                None,  # Diff Totals
-            )
-
-            # add file in chunks
-            self._chunks.append(_file)
-
-        self._totals = None
+            self._files[_file.name] = _file
 
+        self._invalidate_caches()
         return True
 
-    def get(self, filename, _else=None, bind=False):
-        """
-        returns <ReportFile>
-        if not found return _else
-
-        :bind will replace the chunks and bind file changes to the report
-        """
-        _file = self._files.get(filename)
-        if _file is not None:
-            if self._chunks:
-                try:
-                    lines = self._chunks[_file.file_index]
-                except IndexError:
-                    log.warning(
-                        "File not found in chunk",
-                        extra=dict(file_index=_file.file_index),
-                        exc_info=True,
-                    )
-                    lines = None
-            else:
-                # may be tree_only request
-                lines = None
-            if isinstance(lines, ReportFile):
-                return lines
-            report_file = ReportFile(
-                name=filename,
-                totals=_file.file_totals,
-                lines=lines,
-            )
-            if bind:
-                self._chunks[_file.file_index] = report_file
+    def get(self, filename, **kwargs):
```

Comment:
It seems that silently ignoring the `bind` argument (and the `_else` to some extent) _could_ lead to unexpected behavior.

I'm sure this way it's easier to integrate with the rest of the code, but maybe it would be better to fail here if people are using those extra options

Reply:
thats a good point. Python is also a bit annoying with `*args` vs `**kwargs` as well which can cause problems.
regardless of that, I believe the `bind` argument was unused before. I was fixing a bug with that recently as it was completely broken otherwise. So I’m pretty sure noone was using that argument.

I think this PR will need quite a bit of refactoring for downstream code as well, which I haven’t started yet, but will do so soonish.